pkg/table/tables/partition.go

```diff
@@ -128,9 +128,50 @@ func newPartitionedTable(tbl *TableCommon, tblInfo *model.TableInfo) (table.Part
 	if err := initTableIndices(&ret.TableCommon); err != nil {
 		return nil, errors.Trace(err)
 	}
+	origIndices := ret.meta.Indices
+	DroppingDefinitionIndices := make([]*model.IndexInfo, 0, len(origIndices))
+	AddingDefinitionIndices := make([]*model.IndexInfo, 0, len(origIndices))
+	for _, idx := range origIndices {
+		newIdx, ok := pi.DDLChangedIndex[idx.ID]
```

Comment:
Will the `pi.DDLChangedIndex` is nil here?

Reply:
if `pi.DDLChangedIndex` is nil, then there will not be any matches during reading. A nil map will panic on write, but reading is the same as an empty map.

Reply:
Thx, I don't know it before.